File: annotation-processor-testing/src/main/java/com/tickaroo/tikxml/regressiontests/paths/Person.java
package com.tickaroo.tikxml.regressiontests.paths;

import com.tickaroo.tikxml.annotation.PropertyElement;
import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class Person {
  @PropertyElement int id;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Person)) return false;

    Person person = (Person) o;

    return id == person.id;
  }

  @Override public int hashCode() {
    return id;
  }
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/regressiontests/paths/Person.kt
package com.tickaroo.tikxml.regressiontests.paths

import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
open class Person {
    @PropertyElement
    var id: Int = 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Person) return false

        val person = other as Person?

        return id == person!!.id
    }

    override fun hashCode(): Int {
        return id
    }
}


